apache/metron
# METRON-984 Stellar functions to decode encoded fields or strings

This functionality involves ( as a start ) encodings based on the Apache Commons Codec library ( binary )

Support exists for the following encodings:
 * Base32
 * Base32 as hex
 * Base 64
 * Binary
 * Hex

The Stellar functions added:
* LIST_SUPPORTED_ENCODINGS : List the supported encodings
* IS_ENCODING : is this string encoded this way?
* DECODE : decode a string with a given encoding, optionally verify it is in fact encoded that way
* ENCODE: encodes a string with a given encoding

## Contributor Comments
[Please place any comments here.  A description of the problem/enhancement, how to reproduce the issue, your testing methodology, etc.]

### Testing
- Build and tests should run
- run the stellar shell
```bash
cd metron/metron-stellar/stellar-common
mvn -DskipTests package && mvn exec:java -Dexec.mainClass="org.apache.metron.stellar.common.shell.StellarShell"
```
```bash
[Stellar]>>> Functions loaded, you may refer to functions now...

[Stellar]>>> baseString:="METRON is a top level project!"
[Stellar]>>> encoded:=ENCODE(baseString,'BASE64')
[Stellar]>>> encoded
TUVUUk9OIGlzIGEgdG9wIGxldmVsIHByb2plY3Qh
[Stellar]>>> decoded:=DECODE(encoded,'BASE64')
[Stellar]>>> decoded
METRON is a top level project!
[Stellar]>>> IS_ENCODING(encoded,'BASE64')
true
```

### For all changes:


### For code changes:
- [na] Have you included steps to reproduce the behavior or problem that is being changed or addressed?
  ```
  mvn -q clean integration-test install && build_utils/verify_licenses.sh 
  ```


### For documentation related changes:

  ```
  cd site-book
  mvn site
  ```

#### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.
It is also recommended that [travis-ci](https://travis-ci.org) is set up for your personal repository such that your branches are built there before submitting a pull request.
